Mathlib/Topology/Algebra/Module/ModuleTopology.lean

```diff
@@ -0,0 +1,267 @@
+of a module `M` is the quotient topology when `M` is equipped with the module topology.
+
+We will also show the slightly more subtle result that if `M`, `N` and `P` are `R`-modules
+equipped with the module topology and if furthermore `M` is finite as an `R`-module,
+then any bilinear map `M × N → P` is continuous.
+
+As a consequence of this, we deduce that if `R` is a commutative topological ring
+and `A` is an `R`-algebra of finite type as `R`-module, then `A` with its module
+topology becomes a topological ring (i.e., multiplication is continuous).
+
+Other TODOs (not done in the FLT repo):
+
+* The module topology is a functor from the category of `R`-modules
+  to the category of topological `R`-modules, and it's an adjoint to the forgetful functor.
+
+-/
+
+section basics
+
+/-
+This section is just boilerplate, defining the module topology and making
+some infrastructure. Note that we don't even need that `R` is a ring
+in the main definitions, just that it acts on `A`.
+-/
+variable (R : Type*) [TopologicalSpace R] (A : Type*) [Add A] [SMul R A]
+
+/-- The module topology, for a module `A` over a topological ring `R`. It's the finest topology
+making addition and the `R`-action continuous, or equivalently the finest topology making `A`
+into a topological `R`-module. More precisely it's the Inf of the set of
+topologies with these properties; theorems `continuousSMul` and `continuousAdd` show
+that the module topology also has these properties. -/
+abbrev moduleTopology : TopologicalSpace A :=
+  sInf {t | @ContinuousSMul R A _ _ t ∧ @ContinuousAdd A t _}
+
+/-- A class asserting that the topology on a module over a topological ring `R` is
+the module topology. See `moduleTopology` for more discussion of the module topology. -/
+class IsModuleTopology [τA : TopologicalSpace A] : Prop where
+  /-- Note that this should not be used directly, and `eq_moduleTopology`, which takes `R` and `A`
+  explicitly, should be used instead. -/
+  eq_moduleTopology' : τA = moduleTopology R A
+
+theorem eq_moduleTopology [τA : TopologicalSpace A] [IsModuleTopology R A] :
+    τA = moduleTopology R A :=
+  IsModuleTopology.eq_moduleTopology' (R := R) (A := A)
+
+/-- Scalar multiplication `• : R × A → A` is continuous if `R` is a topological
+ring, and `A` is an `R` module with the module topology. -/
+theorem ModuleTopology.continuousSMul : @ContinuousSMul R A _ _ (moduleTopology R A) :=
+  /- Proof: We need to prove that the product topology is finer than the pullback
+     of the module topology. But the module topology is an Inf and thus a limit,
+     and pullback is a right adjoint, so it preserves limits.
+     We must thus show that the product topology is finer than an Inf, so it suffices
+     to show it's a lower bound, which is not hard. All this is wrapped into
+     `continuousSMul_sInf`.
+  -/
+  continuousSMul_sInf <| fun _ _ ↦ by simp_all only [Set.mem_setOf_eq]
```

Comment:
This is a long comment for a very small amount of proof!

Reply:
Yes, and in my opinion more of mathlib should look like this!